### ISSUE TITLE
Report of unparseable Sections

### ISSUE THREAD

**Issue by dominicmeyer** [sqlfluff/sqlfluff#5261]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following mysql lines are all marked as unparseable, even though they should be correct.

### Expected Behaviour

Showing either an error, if the format is incorrect or success

### Observed Behaviour

sql statements seem to be unknown to sqlfluff

### How to reproduce

lint a sql-file, with the following lines:
```sql
# unparsable section: CHARACTER SET utf8mb4
CREATE DATABASE IF NOT EXISTS example CHARACTER SET utf8mb4;
# parseable
USE example;
# unparsable section: CREATE USER 'user' IDENTIFIED BY 'password'
CREATE USER 'user' IDENTIFIED BY 'password';
# unparsable section: GRANT ALL PRIVILEGES ON example.* TO 'user'
GRANT ALL PRIVILEGES ON example.* TO 'user';
# unparsable section: FLUSH PRIVILEGES
FLUSH PRIVILEGES;

```

### Dialect

mysql

### Version

Unknown, the current version used in [super-linter](https://github.com/super-linter/super-linter)

### Configuration

[sqlfluff]
dialect = mysql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
